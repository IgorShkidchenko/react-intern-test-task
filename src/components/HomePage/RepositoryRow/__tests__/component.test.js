import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import RepotisoryRowComponent from "../component";

configure({ adapter: new Adapter() });

describe("<RepotisoryRowComponent />", () => {
  const clickFn = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RepotisoryRowComponent
      repository={{ name: "name", stargazers_count: 1 }}
      handleClick={clickFn}
    />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleClick on X click", () => {
    wrapper.find("span").simulate("click");

    expect(clickFn).toHaveBeenCalled();
  });
});
