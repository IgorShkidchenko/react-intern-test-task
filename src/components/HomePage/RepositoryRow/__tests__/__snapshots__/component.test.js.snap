// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepotisoryRowComponent /> renders correctly 1`] = `
<tr>
  <td>
    <Link
      to={
        Object {
          "pathname": "/details",
          "state": Object {
            "repository": Object {
              "name": "name",
              "stargazers_count": 1,
            },
          },
        }
      }
    >
      name
       
      1
      <span
        onClick={[MockFunction]}
        onKeyDown={[MockFunction]}
        role="button"
        tabIndex={0}
      >
          Ð¥
      </span>
    </Link>
  </td>
</tr>
`;
